packages/muelu/src/Graph/UncoupledAggregation/MueLu_AggregationPhase2aAlgorithm_kokkos_def.hpp

```diff
@@ -165,13 +165,13 @@ namespace MueLu {
                                         if(graph.isLocalNeighborVertex(neigh) &&
                                            aggStat(neigh) == READY &&
                                            aggSize < maxNodesPerAggregate) {
```

Comment:
Unrelated note: once `aggSize` hits `maxNodesPerAggregate` in the loop above, this condition is always false.

Reply:
Changed the logic so that a partial aggregate is now form as it was the case in the non-refactored code.